Sources/WebSocketKit/PMCE.swift

```diff
@@ -0,0 +1,753 @@
+                              rsv1: maskedFrame.rsv1,
+                              rsv2: maskedFrame.rsv2,
+                              rsv3: maskedFrame.rsv3,
+                              opcode: maskedFrame.opcode,
+                              maskKey: nil,
+                              data: unmaskedData,
+                              extensionData: maskedFrame.extensionData)
+    }
+
+    ///
+    private let logger = Logger(label: "PMCE")
+    private let _logging:NIOLockedValueBox<Bool>
+    private let _enabled:NIOLockedValueBox<Bool>
+
+    // Converts windowBits to size of window.
+    private static func sizeFor(bits:UInt8) -> Int32 {
+        2^Int32(bits)
+    }
+    
+    private func pad(buffer:ByteBuffer) -> ByteBuffer {
+        logger.trace("padding buffer with \(paddingOctets)")
+        var mutbuffer = buffer
+        mutbuffer.writeBytes(paddingOctets)
+        return mutbuffer
+    }
+
+    private func unpad(buffer:ByteBuffer) -> ByteBuffer {
+        logger.trace("unpadding \(buffer)")
+        return buffer.getSlice(at: 0, length: buffer.readableBytes - 4) ?? buffer
+    }
+    
+    // client compression uses this
+    private func makeMaskKey() -> WebSocketMaskingKey? {
+        switch extendedSocketType {
+        
+        case .client:
+            let mask = WebSocketMaskingKey.random()
+            logger.trace("PMCE: created mask key \(mask) .")
+            return mask
+        case .server:
+            logger.trace("PMCE: mask key is nil due to extending server socket type.")
+            return nil
+        }
+    }
+    
+    // Box for compressor to conform to Sendable.
+    private let compressorBox:NIOLoopBoundBox<NIOCompressor?>
+    
+    // 4 bytes used for compress and decompress when context takeover is being used.
+    private let paddingOctets:[UInt8] = [0x00, 0x00, 0xff, 0xff]
+
+    // Box for compressor to conform to Sendable.
+    private let decompressorBox:NIOLoopBoundBox<NIODecompressor?>
+    
+    // sometimes see internalError when server gets ctrl-c'd. I think it is related to the issues Ive seeen with stopping the server in general via ctrl-c.
+    deinit {
+       stopStreams()
+    }
+    
+}
+
+// any ideas to make it prettier or even add colors are welcome, reviiewers.
+extension PMCE: CustomStringConvertible {
+    public var description: String {
+        """
+        enabled : \(enabled),
+        extendedSocketType : \(self.extendedSocketType),
+        serverConfig : \(serverConfig),
+        clientConfig : \(clientConfig)
+        """
+    }
+}
+
+extension PMCE.PMCEConfig: Equatable {
+    public static func == (lhs: PMCE.PMCEConfig,
+                           rhs: PMCE.PMCEConfig) -> Bool {
+        return lhs.headerParams() == rhs.headerParams()
+    }
+}
+
+extension PMCE.PMCEConfig: CustomDebugStringConvertible {
+    public var debugDescription: String {
+        "PMCEConfig {config:\(deflateConfig)}"
+    }
+}
```

Comment:
It is nonsensical for debug description to provide less information than regular description.